src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/UpToDate/UpToDateCheckStatePersistence.cs

```diff
@@ -125,9 +125,9 @@ async Task<string> GetCacheFilePathAsync(CancellationToken cancellationToken)
             }
         }
 
-        public void StoreState(string projectPath, IImmutableDictionary<string, string> configurationDimensions, int itemHash, DateTime itemsChangedAtUtc)
+        public async Task StoreStateAsync(string projectPath, IImmutableDictionary<string, string> configurationDimensions, int itemHash, DateTime itemsChangedAtUtc)
         {
```

Comment:
It feels like Cancellation representing project unload should be plumbed through here in a future PR, quick opens and closes are going to block on this completing.

Reply:
Fixed in 27e25428330438da6929295b101725d9ad98c0da.